### ISSUE TITLE
Favicons sometimes incorrect

### ISSUE THREAD

**Issue by jonas-w** [SimonHalvdansson/Harmonic-HN#108]
I don't know why exactly sometimes the favicons are weird, but for example minecraft.wiki shows the mastodon logo https://api.faviconkit.com/minecraft.wiki/80 but if you use a slightly lower resolution https://api.faviconkit.com/minecraft.wiki/76 it returns the correct one. Maybe use 76 instead of 80, as 76 is some kind of "standard" size and if you put in a size which does not exist, faviconkit will show you a larger size if that does exist. e.g. https://api.faviconkit.com/minecraft.wiki/80 returns 144 ⇾ https://api.faviconkit.com/minecraft.wiki/144

I also very often see the archive.org logo on entries that definitely have nothing to do with archive.org
You could also don't specify a size at all https://api.faviconkit.com/minecraft.wiki/ which would in this case also return the correct favicon.

**Comment by SimonHalvdansson** [comment]
Hmm interesting, yeah perhaps leaving the field blank is the best way to do things. When I Googled this a bit I realized there are some alternatives available also:

https://icons.duckduckgo.com/ip3/openculture.com.ico
https://www.google.com/s2/favicons?domain=openculture.com&sz=128

which perhaps are faster than the current API - although they both have GitHub favicons that look worse then the current one so it would hurt a bit to change. With the regards to the archive.org thing, I assume the Google one is the most "stable" option. I tried them all right now and faviconkit was the most up-to-date actually.

For now, I'll remove the /80 thing - let me know if you have any more input on this.

**Comment by SimonHalvdansson** [comment]
Yeah those things are unfortunate - for pearl.hifi.com both Google and DDG work properly for me right now. I'm thinking letting the user set this (with faviconkit as the default) maybe isn't so bad of a solution.

**Comment by SimonHalvdansson** [comment]
https://github.com/SimonHalvdansson/Harmonic-HN/commit/6ebd0fea50e12cc1b5b56ca93bbe4898cfa4f4fa

**Comment by SimonHalvdansson** [comment]
Google needed some size to not load the 16x16 icon (which is too small) but now I investigated it a bit more and Google needs you to indicate the exact correct size and therefore I changed it to 128x128 (which after 16x16 seems to be the most common one). Also the hardcoded 80x80 resizing is replaced by a dynamic 17dpx17dp in pixels.

Also the reason the favicon is 17dp is that I think it looked the best - I usually follow 4dp increments but with these precise layout things it is more about looking reasonable in size next to the text.
